### PR TITLE
feat: Add Base support to Uniswap widget

### PR DESCRIPTION
**Pull request by abarthell**
Add Base support to Uniswap widget. 

Tested swapping ETH to USDC on Base: https://basescan.org/tx/0xb230b5039555c3087b61906555b1b9495395cb60486b68f9871f791505429789

Screen Shot 2023-09-21 at 3 06 15 PM
Screen Shot 2023-09-21 at 3 13 15 PM

### CODE REVIEW

File: src/constants/chainInfo.ts [hunk 1]
```diff
@@ -237,6 +238,19 @@ const CHAIN_INFO: ChainInfoMap = {
     color: '#F0B90B',
     backgroundColor: '#F0B90B',
   },
+  [SupportedChainId.BASE]: {
+    networkType: NetworkType.L2,
+    blockWaitMsBeforeWarning: ms`10m`,
```
Comment (danyalprout) [inline src/constants/chainInfo.ts:243]: It maybe worth making this match optimism:

```
    blockWaitMsBeforeWarning: ms`25m`,
```

File: src/constants/jsonRpcEndpoints.ts [hunk 2]
```diff
@@ -86,4 +86,14 @@ export const JSON_RPC_FALLBACK_ENDPOINTS: Record<SupportedChainId, string[]> = {
     'https://bsc-dataseed4.defibit.io',
     'https://rpc.ankr.com/bsc',
   ],
+  [SupportedChainId.BASE]: [
+    // "Safe" URLs
+    'https://mainnet.base.org',
+    'https://developer-access-mainnet.base.org/',
```
Comment (danyalprout) [inline src/constants/jsonRpcEndpoints.ts:92]: Lets remove `developer-access-mainnet.base.org`, it resolves to the same hosts as `mainnet.base.org` and at some point we may remove it.

**Comment by just-toby** [comment]
@abarthell assuming you were able to test locally, get quotes and execute trades - please mark this PR ready for review when you want us to review it !

**Comment by abarthell** [comment]
> @abarthell assuming you were able to test locally, get quotes and execute trades - please mark this PR ready for review when you want us to review it !

@just-toby I don't think this PR will work until USDC on Base is added [here](https://github.com/Uniswap/interface/blob/main/src/constants/tokens.ts#L77-L81) (instead of USDbC). Should I also make a change there first?

**Comment by just-toby** [comment]
@abarthell that's a token constant in our `interface` repo - it doesn't affect the widget. we can/should update it, but i don't think it's related to this PR.

**Comment by abarthell** [comment]
@just-toby I tested the swap, quote, etc (shown in the screenshots in the PR description), but I am getting this error below when I select a destination currency to swap to (even though I added the multicall address in the PR); do you know if I am missing something?
Screen Shot 2023-09-21 at 3 06 32 PM

File: package.json [hunk 3]
```diff
@@ -67,7 +67,7 @@
     "@uniswap/redux-multicall": "^1.1.8",
     "@uniswap/router-sdk": "^1.6.0",
     "@uniswap/sdk-core": "^4.0.6",
-    "@uniswap/smart-order-router": "^3.13.7",
+    "@uniswap/smart-order-router": "^3.16.25",
```
Comment (just-toby) [inline package.json:70]: nice, i was about to comment but you found it 👍

File: src/constants/chains.ts [hunk 4]
```diff
@@ -21,6 +21,8 @@ export enum SupportedChainId {
   CELO_ALFAJORES = 44787,
 
   BNB = 56,
+
+  BASE = 8453,
```
Comment (just-toby) [inline src/constants/chains.ts:25]: at some point our team might remove this enum entirely, like i started to do in #615 but for now this works

**Comment by just-toby** [comment]
@abarthell please run `yarn deduplicate` - i'm seeing the lints and tests passing locally, so i don't think we need to worry about those checks. probably just need to rerun or fix the github action

**Comment by abarthell** [comment]
> @abarthell please run `yarn deduplicate` - i'm seeing the lints and tests passing locally, so i don't think we need to worry about those checks. probably just need to rerun or fix the github action

@just-toby Sounds good! just ran it locally and also bumped the deps it was erroring on

**Comment by just-toby** [comment]
the test suite is only failing in your fork because you don't have the  `JSON_RPC_PROVIDER`  env variable set. it's passing for me and it should pass on `main` once we merge

**Review by just-toby — APPROVED** [review]

**Comment by abarthell** [comment]
> the test suite is only failing in your fork because you don't have the `JSON_RPC_PROVIDER` env variable set. it's passing for me and it should pass on `main` once we merge
> 
> edit: i forgot the check is required. so we don't break protocol here, do you mind setting the env variable so we can re-run the tests and get all green ? 

@just-toby Ahh I see, good catch. I just set `JSON_RPC_PROVIDER = https://mainnet.infura.io` in the fork env vars

**Comment by just-toby** [comment]
hmm @abarthell  i think i was wrong and it's _our_ repo that has the wrong value for this variable. i'll get it updated and make sure your PR gets merged. (might be tomorrow since i don't have permissions to edit the settings in this repo)

**Comment by abarthell** [comment]
@just-toby Can we try re-running it one more time? I think I had the wrong variable before actually

**Comment by abarthell** [comment]
@just-toby can you help upgrade [the latest npm version](https://www.npmjs.com/package/@uniswap/widgets) of the widget as well? 😁

Also, here's two more PRs to add Base USDC on the interface (instead of USDbC) and the corresponding assets: 
- https://github.com/Uniswap/interface/pull/7385
- https://github.com/Uniswap/assets/pull/38

**Pull request merged**